### ISSUE TITLE
Source-Sink and Autoconnexion Jack

### ISSUE THREAD

**Issue by CarlBcx** [Houston4444/RaySession#182]
Good morning,

I discovered RaySession recently and little by little I'm starting to master it.
I used Qjackctl before, then I tested Cadence which caused a lot of problems.
So I definitely adopted RaySession.
This is the kind of setup I'm using.
![RaySession](https://github.com/Houston4444/RaySession/assets/101471818/0106afb0-4495-4128-94b2-657cb198f3bc)

First of all, I would like to thank you for this fantastic tool which saves me precious time and which is remarkably well thought out.

I manage to manage for almost everything because I use a lot of command line tools and scripts, but I am stuck on two points.

1. How to have the Source and the Sink that I had in qjackctl and which allow to connect non-jack compatible software or which are on other sound cards.

2. Disable automatic logout.
Indeed some at the opening connect to other peripherals, in addition to the settings kept.
For example in my case, Hydrogen always connects to the sound output while it should only connect to Jamulus.
In Qjackctl there is an option "Ignore all auto-login requests".
As I run Jack on the command line I thought I'd find that as an argument but no, it's not in the doc.
I've done tons of research but haven't found it yet.

So if you have an idea....

Best regards
Carl

**Comment by Houston4444** [comment]
Hi !

> First of all, I would like to thank you for this fantastic tool which saves me precious time and which is remarkably well thought out.

Thanks, always nice to hear ;).

> I used Qjackctl before, then I tested Cadence which caused a lot of problems.
So I definitely adopted RaySession.

Mmmh, here you make a confusion, Cadence and QJackCtl are used for the same task (manage JACK settings), RaySession is not done for that. There are no JACK settings in RaySession. Personally I use Cadence + RaySession, you can perfectly use command lines + RaySession. Cadence and QJackCtl reliability will mainly depends on your distribution, here with Librazik, absolutely no problem.

> How to have the Source and the Sink that I had in qjackctl and which allow to connect non-jack compatible software or which are on other sound cards.

PulseAudio bridges ? I personally launch them with Cadence, you can also use the script given by QJackCtl interface:
`pactl load-module module-jack-sink channels=2 connect=no && pactl load-module module-jack-source && echo "set-default-sink jack_out" | pacmd && echo "set-default-source jack_in" | pacmd && a2j -e`

> Disable automatic logout.
Indeed some at the opening connect to other peripherals, in addition to the settings kept.
For example in my case, Hydrogen always connects to the sound output while it should only connect to Jamulus.
In Qjackctl there is an option "Ignore all auto-login requests".
As I run Jack on the command line I thought I'd find that as an argument but no, it's not in the doc.
I've done tons of research but haven't found it yet.

There is an option in Hydrogen for that in **Preferences -> Audio System**. with `jackd --help`, I see you can use ` -autoconnect e` option to jackd for that.

**Comment by CarlBcx** [comment]
Yes I prefer to use only Raysession and run jack by command line.

About the sink and source, finally I need only sink to send music into Jamulus.
So I created a Hack into Raysession with a bash script who has `pactl load-module module-jack-sink channels=2 connect=no` into it.
It's work fine.

I tryed `jack_control eps self-connect-mode e` and it's perfect.
I didn't have the idea to change that after jack is started.

Hydrogen option changed too.

Thanks again for all !

Carl
